### DISCUSSION TITLE
View query parameters in imported OpenAPI definitions

### DISCUSSION THREAD

**Discussion by majadinjar** [mockoon/mockoon#692] [Ideas] [answered]
Only one thing is missing so that Mockoon became perfect tool for OpenAPI mocking. I can't find list of possible query parameters.

It would be great if you add list of possible query params in every api definition, you can read it from parameters attribute in OpenAPI documentation:
```
"parameters": 
[ {
      "name": "name",
      "in": "query",
      "description": "name",
      "required": false,
      "style": "form",
      "schema": {
          "type": "string"
      }
 } ]
```
I think "name", "description" (maybe on hover), "required" (important one), and I'm not sure if any other attribute is needed.

If I missed something and that's already implemented please help me to find it.

Thanks.

**Comment by 255kb** [comment, marked as answer]
Mockoon is not an API design tool and wasn't built with this in mind. 
The main consequences are:
- Mockoon only takes care of the response. There is currently no way to describe what the request should be, like for the query params. There is also no plan to update the UI to include the request definition.
- the OpenAPI support is only a partial one. Some OpenAPI concepts have no equivalent in Mockoon and vice versa (e.g. Mockoon's rules system), resulting in possible information loss when importing from or exporting to an OAS file. 

I think Mockoon's strength is to be a good API mocking tool rather than an API design tool with average mocking capabilities. I am not sure that adding API design features that don't serve the mocking side would benefit us.

**Reply by majadinjar** [reply]
> Mockoon is not an API design tool and wasn't built with this in mind. The main consequences are:
> 
> * Mockoon only takes care of the response. There is currently no way to describe what the request should be, like for the query params. There is also no plan to update the UI to include the request definition.
> * the OpenAPI support is only a partial one. Some OpenAPI concepts have no equivalent in Mockoon and vice versa (e.g. Mockoon's rules system), resulting in possible information loss when importing from or exporting to an OAS file.
> 
> I think Mockoon's strength is to be a good API mocking tool rather than an API design tool with average mocking capabilities. I am not sure that adding API design features that don't serve the mocking side would benefit us.

I can't agree that API design features don't serve mocking. Just think about case where you need to get list of data for page 1, page size 10, and sort by name. It is very nice feature that you can read it from query params and put it in response. That helped frontend team to easily mock any data table with pagination and that mocked response isn't just fixed one.

Currently we have situation where our backend team has less number of members than a frontend team and we can't work in parallel. So backend team wrote OpenAPI definitions and Mockoon is almost all we need to develop frontend part of the app. Later when API part is done, we just change api context. 

The only thing currently missing is that team can't see exact possible query parameters they are able to send if they don't look at OpenAPI definition.

My opinion is that Mockoon should go exactly in direction to take care that mocked data is as much real as it could be. And reading query params and maybe type definitions from OpenAPI is not such a big feature to add and it can bring great value. Just my opinion :)

**Reply by 255kb** [reply]
Sure, I understand what you mean :) 
Still, I think there are other features that should be prioritized over this one due to their popularity. But feel free to open an issue describing what you would expect from such feature and we can see if it gathers interest from other users.

**Reply by bewareilive** [reply]
> Sure, I understand what you mean :) Still, I think there are other features that should be prioritized over this one due to their popularity. But feel free to open an issue describing what you would expect from such feature and we can see if it gathers interest from other users.

I have submitted the same question, and I find that someone has submitted it before me.This is the third one.
We can use without parameters, but we need to know which request parameters are needed to get the data. This is just as important as knowing the IP, port, method name, etc. of the request.
